dataloader.py

```diff
@@ -14,35 +17,48 @@ def __init__(
         label_folder: Union[str, pathlib.Path],
     ):
         super().__init__()
-        self.seqs = self.read_seq(seq_folder)
-        self.labels = self.read_labels(label_folder)
+        self.targets = self.read_FASTA_file(seq_folder, label_folder)
 
-    def read_seqs(self, seq_folder: Union[str, pathlib.Path]):
-        seq_results = []
-        for seq_file in listdir(seq_folder):
-            if isfile(join(seq_folder, seq_file)) and seq_file.endswith(".csv"):
-                seq_results.append(self.read_seq_file(join(seq_folder, seq_file)))
+    def read_FASTA_file(
+        self,
+        seq_filename: Union[str, pathlib.Path],
+        label_folder: Union[str, pathlib.Path],
+    ):
+        print("generate fasta sequence")
+        targets = []
+        genes = Fasta(seq_filename)
+        labels = self.read_labels(label_folder)
+        for chr, len in chr_length.items():
+            for i in tqdm(range(0, len, 1500)):
+                sample = genes[chr][i : i + 2000].seq
+                label = self.find_corresponding_label(i, i + 2000, labels[chr])
+                targets.append((sample, label))
+        return targets
 
-        # concatenate all chr content
-        seq_arrays = seq_results[0]
-        for i in range(1, len(seq_results)):
-            seq_arrays = np.concatenate((seq_arrays, seq_results[i]), axis=0)
-        return seq_arrays
+    def find_corresponding_label(self, seq_start: int, seq_end: int, labels):
+        results = []
+        for label in labels:
+            align_start, align_end = int(label[0]), int(label[1])
+            if align_start > align_end:
+                continue
+            if seq_start <= align_start and align_end <= seq_end:
```

Comment:
For now, I skip the situation about the information on the reverse strand, it seems like FASTA file only contains the information of forward strand, should we consider it?

Reply:
Hi Yantong. **Let me check the data you're using and I'll let you know.**
I would say that, for the time being, you can ignore it as this is still sort of a proof of concept (in the sense that there are more functionalities missing, such as different types of repeats). To extract the reverse complement from a fasta file is really simple, in any case.

Reply:
Hi again, just wanted to double check, but as suspected, **you should consider the reverse strand**.
As I said, this should be really easy to fix (a couple of one-line conditionals came to mind, even). Let me know if I can be of further assistance.
Once more, if you want to finish the proof of concept before, I guess that would be ok... But I don't see a reason for waiting.

Reply:
Thanks for your suggestion! It solves my confusion. I will add it to the dataloador.:D